torch/functional.py

```diff
@@ -1406,5 +1408,108 @@ def _lu_no_infos(A, pivot=True, get_infos=False, out=None):
     func_name='lu')
 lu.__doc__ = _lu_impl.__doc__
 
+def clamp(input, min=None, max=None, out=None):
+    r"""
+    Clamps all elements in :attr:`input` into the range `[` :attr:`min`, :attr:`max` `]` and returns a tensor:
+
+    .. math::
+        y_i = \min(\text{max}_i, \min(\text{min}_i, x_i))
+
+    The output dtype is same as :attr:`input` if :attr:`min`/:attr:`max` arguments are scalars, otherwise if the 
+    arguments are tensors then the output dtype uses standard type promotion rules.
+
+    Args:
+        input (Tensor): the input tensor
+        min (Tensor or Number): lower-bound of the range to be clamped to
+        max (Tensor or Number): upper-bound of the range to be clamped to
+        out (Tensor, optional): the output tensor.
+
+    Example::
+
+        >>> a = torch.randn(4)
+        >>> a
+        tensor([-1.7120,  0.1734, -0.0478, -0.0922])
+        >>> torch.clamp(a, min=-0.5, max=0.5)
+        tensor([-0.5000,  0.1734, -0.0478, -0.0922])
+        >>>
+        >>> min_tensor = torch.tensor([-0.5, -1.0, 0.0, -0.5])
+        >>> max_tensor = torch.tensor([1.0, 1.0, 0.0, -0.1])
+        >>> torch.clamp(a, min=min_tensor, max=max_tensor)
+        tensor([-0.5000,  0.1734,  0.0000, -0.1000])
+
+    .. function:: clamp(input, *, min, out=None) -> Tensor
+
+    Clamps all elements in :attr:`input` to be larger or equal to :attr:`min`.
+
+    The output dtype is same as attr:`input` if :attr:`min` argument is a scalar, otherwise if :attr:`min` is a tensor
+    then the output dtype uses standard type promotion rules.
+
+    Args:
+        input (Tensor): the input tensor
+        min (Tensor or Number): minimal value of each element in the output
+        out (Tensor, optional): the output tensor.
+
+    Example::
+
+        >>> a = torch.randn(4)
+        >>> a
+        tensor([-0.0299, -2.3184,  2.1593, -0.8883])
+        >>> torch.clamp(a, min=0.5)
+        tensor([ 0.5000,  0.5000,  2.1593,  0.5000])
+        >>>
+        >>> min_tensor = torch.tensor([-0.5, -1.0, 0.0, -0.5])
+        >>> torch.clamp(a, min=min_tensor)
+        tensor([-0.0299, -1.0000,  2.1593, -0.5000])
+
+    .. function:: clamp(input, *, max, out=None) -> Tensor
+
+    Clamps all elements in :attr:`input` to be smaller or equal :attr:`max`.
+
+    The output dtype is same as attr:`input` if :attr:`max` argument is a scalar, otherwise if :attr:`max` is a tensor
+    then the output dtype uses standard type promotion rules.
+
+    Args:
+        input (Tensor): the input tensor
+        max (Tensor or Number): maximal value of each element in the output
```

Comment:
Keyword args:
max (Tensor or Number): maximal value of each element in the output
out (Tensor, optional): the output tensor (<- no period for consistency)